app/Model/Test.php

```diff
@@ -302,11 +312,65 @@ public function GetTestMeasurementCollection()
     }
 
     /**
-     * @return
+     * @return LabelCollection
+     */
+    public function GetLabelCollection()
+    {
+        if (!$this->LabelCollection) {
+            $this->LabelCollection = new LabelCollection();
+            foreach ($this->Labels as $label) {
+                $this->LabelCollection->add($label);
+            }
+        }
+        return $this->LabelCollection;
+    }
+
+    /**
+     * @return double
      */
     public function GetExecutionTime()
     {
         $buildTest = $this->GetBuildTest();
         return (double)$buildTest->Time;
     }
+
+    /**
+     * @return bool
+     */
+    public function HasFailed()
+    {
+        return $this->GetStatus() === self::FAILED;
+    }
+
+    /**
+     * @return bool
+     */
+    public function HasNotRun()
+    {
+        return $this->GetStatus() === self::NOTRUN;
+    }
+
+    /**
+     * @return bool
+     */
+    public function HasPassed()
+    {
+        return $this->GetStatus() === self::PASSED;
+    }
```

Comment:
"Has" is kind of an awkward verb for test status.  How about IsPassed, IsFailed, IsNotRun instead?